pkg/util/time.go

```diff
@@ -113,6 +113,57 @@ func NewDisableableTicker(interval time.Duration) (func(), <-chan time.Time) {
 	return func() { tick.Stop() }, tick.C
 }
 
+// NewVariableTicker wrap time.Ticker to Reset() the ticker with the next duration (picked from
+// input durations) after each tick. The last configured duration is the one that will be preserved
+// once previous ones have been applied.
+//
+// Returns a function for stopping the ticker, and the ticker channel.
```

Comment:
`ticker.Stop` function can be called multiple times. Custom stop function cannot be called multiple times. We should document expected behaviour here.

Reply:
Very good point. I've fixed it in https://github.com/grafana/mimir/pull/7486/commits/53c7a3cbc58ba162c507a0e7a13bd6fc1907fd4f